../tcg/tcg-op.c:1750:28: runtime error: signed integer overflow: 9223372036854775807 + 1 cannot be represented in type 'int64_t' (aka 'long')

fixes a signed overflow breaking TCG under
certain conditions, esp. with qemu-user emulation (causing
internal SIGILL)

source: https://github.com/chimera-linux/cports/blob/master/main/qemu/patches/tcg-overflow.patch

--- a/tcg/tcg-op.c
+++ b/tcg/tcg-op.c
@@ -1747,7 +1747,7 @@ void tcg_gen_andi_i64(TCGv_i64 ret, TCGv_i64 arg1, int64_t arg2)
          * which does not require matching operands.  Other backends can
          * trivially expand the extract to AND during code generation.
          */
-        if (!(arg2 & (arg2 + 1))) {
+        if (!((uint64_t)arg2 & ((uint64_t)arg2 + 1))) {
             unsigned len = ctz64(~arg2);
             if (TCG_TARGET_extract_valid(TCG_TYPE_I64, 0, len)) {
                 tcg_gen_extract_i64(ret, arg1, 0, len);
